egs/wsj/s5/steps/cleanup/segment_long_utterances.sh

```diff
@@ -0,0 +1,389 @@
+Usage: $0 [options] <model-dir> <lang> <data-in> <segmented-data-out> <work-dir>
+ e.g.: $0 exp/tri3 data/train_unsegmented data/train_segmented exp/tri3_segmentation
+This script performs segmentation of the data in <data-in>, writing the segmented
+data (with a segments file) to <segmented-data-out>.  The purpose of this script is
+to divide up the input data (which may consist of long recordings such as television
+shows or audiobooks) into segments which are of manageable length for further
+processing, along with the portion of the transcript that seems to match each segment.
+The output data is not necessarily particularly clean; you are advised to 
+run steps/cleanup/clean_and_segment_data.sh on the output in order to further
+clean it and eliminate data where the transcript doesn't seem to match.
+EOF
+    exit 1
+fi
+
+srcdir=$1
+lang=$2
+data=$3
+out_data=$4
+dir=$5
+
+for f in $data/feats.scp $data/cmvn.scp $srcdir/tree \
+  $srcdir/final.mdl $srcdir/cmvn_opts; do
+  if [ ! -f $f ]; then
+    echo "$0: Could not find file $f"
+    exit 1
+  fi
+done
+
+data_id=`basename $data`
+mkdir -p $dir
+
+data_uniform_seg=$dir/${data_id}_uniform_seg
+
+# First we split the data into segments of around 30s long, on which 
+# it would be possible to do a decoding.
+if [ $stage -le 1 ]; then
+  echo "$0: Splitting data directory $data into uniform segments..."
+
+  utils/data/get_utt2dur.sh $data
+
+  # TODO: Write this script
+  # utils/data/get_uniform_subsegments.py \
+  #   --max-segment-duration=$max_segment_duration \
+  #   --overlap-duration=$overlap_duration \
+  #   $data/segments > $dir/uniform_sub_segments
+
+  utils/data/get_segments_for_data.sh $data > $data/segments
+
+  cat $data/utt2spk | awk 'BEGIN{i=1} {print $1" "i; i++}' > $data/utt2label
+
+  segmentation-init-from-segments --shift-to-zero=true \
+    --utt2label-rspecifier=ark,t:$data/utt2label $data/segments ark:- | \
+    segmentation-post-process \
+      --max-segment-length=`perl -e "print $max_segment_duration * 100"` \
+      --overlap-length=`perl -e "print $overlap_duration * 100"` \
+      ark:- ark:- | \
+    segmentation-to-segments --single-speaker \
+      ark:- ark:/dev/null $dir/uniform_sub_segments
+  #- | \
+  #  utils/int2sym.pl -f 2 $data/utt2label > $dir/uniform_sub_segments
+
+  awk '{print $1" "$2}' $dir/uniform_sub_segments > $dir/new2old_utts
+fi
+
+if [ $stage -le 2 ]; then
+  utils/data/subsegment_data_dir.sh $data $dir/uniform_sub_segments $data_uniform_seg
+
+  # make frame-overlap slightly larger so that the number of frames decreases 
+  # by perhaps at most 1 to fix some round-off errors.
+  rm $data/reco2num_frames $data/reco2dur || true
+  utils/data/get_reco2num_frames.sh --frame-shift 0.01 --frame-overlap 0.0151 \
+    $data
+
+  awk '{print $1" "$2}' $data_uniform_seg/segments | \
+    utils/apply_map.pl -f 2 $data/reco2num_frames > \
+    $data_uniform_seg/utt2max_frames
+
+  utils/data/get_subsegment_feats.sh $data/feats.scp 0.01 0.015 \
```

Comment:
why are you adding the script utils/data/get_subsegment_feats.sh?
steps/cleanup/split_long_utterance.sh already takes care of creating the feats.scp using ranges.
If there is some problem with the output of utils/data/subsegment_data_dir.sh it would be better to fix it there.